hibernate-osgi/src/main/java/org/hibernate/osgi/OsgiClassLoader.java

```diff
@@ -212,4 +214,18 @@ public void clear() {
 		resourceCache.clear();
 	}
 
+	public void removeBundle(Bundle bundle) {
```

Comment:
Yes, that's doable.  The reason was that I modified the addBundle() to call addClassLoader() instead (the bundles field is currently not used anymore).
I can turn the bundles Set into a Map.